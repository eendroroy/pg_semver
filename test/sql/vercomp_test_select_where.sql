LOAD 'vercomp';

CREATE TABLE ver_test(version VER);

INSERT INTO ver_test VALUES ('0.0.0'), ('1.0.0'), ('2.0.0-beta1'), ('2.0.0-rc1'),('2.10.0-beta0'), ('2.2.0-alpha'),  ('3.0.0');

SELECT * FROM ver_test WHERE version =  '1.0.0'::VER;
SELECT * FROM ver_test WHERE version >  '1.0.0'::VER;
SELECT * FROM ver_test WHERE version <  '1.0.0'::VER;
SELECT * FROM ver_test WHERE version <> '1.0.0'::VER;

SELECT * FROM ver_test WHERE version =  '2.0.0-alpha'::VER;
SELECT * FROM ver_test WHERE version >  '2.0.0-alpha'::VER;
SELECT * FROM ver_test WHERE version <  '2.0.0-alpha'::VER;
SELECT * FROM ver_test WHERE version <> '2.0.0-alpha'::VER;


SELECT * FROM ver_test WHERE version > '2.9-beta1'::VER;

DROP TABLE IF EXISTS ver_test;