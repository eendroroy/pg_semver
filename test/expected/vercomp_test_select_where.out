LOAD 'vercomp';
CREATE TABLE versions(version VERSION);
INSERT INTO versions VALUES ('1.0.0'), ('0.0.0'), ('1.1.0'), ('1.2.0'), ('1.23.0'), ('1.23.9'), ('2.9.0'), ('2.0.12'),
                            ('2.0.0-alpha'), ('3.0.1'), ('3.0.2'), ('3.0.4-beta2'), ('3.0.4-alpha'), ('3.0.4-rc1'),
                            ('1.5.0'), ('1.05.0'), ('1.005.0');
SELECT * FROM versions WHERE version =  '1.5.0';
 version 
---------
 1.5.0
 1.05.0
 1.005.0
(3 rows)

SELECT * FROM versions WHERE version != '1.5.0';
   version   
-------------
 1.0.0
 0.0.0
 1.1.0
 1.2.0
 1.23.0
 1.23.9
 2.9.0
 2.0.12
 2.0.0-alpha
 3.0.1
 3.0.2
 3.0.4-beta2
 3.0.4-alpha
 3.0.4-rc1
(14 rows)

SELECT * FROM versions WHERE version <> '1.5.0';
   version   
-------------
 1.0.0
 0.0.0
 1.1.0
 1.2.0
 1.23.0
 1.23.9
 2.9.0
 2.0.12
 2.0.0-alpha
 3.0.1
 3.0.2
 3.0.4-beta2
 3.0.4-alpha
 3.0.4-rc1
(14 rows)

SELECT * FROM versions WHERE version >  '1.5.0';
   version   
-------------
 1.23.0
 1.23.9
 2.9.0
 2.0.12
 2.0.0-alpha
 3.0.1
 3.0.2
 3.0.4-beta2
 3.0.4-alpha
 3.0.4-rc1
(10 rows)

SELECT * FROM versions WHERE version >= '1.5.0';
   version   
-------------
 1.23.0
 1.23.9
 2.9.0
 2.0.12
 2.0.0-alpha
 3.0.1
 3.0.2
 3.0.4-beta2
 3.0.4-alpha
 3.0.4-rc1
 1.5.0
 1.05.0
 1.005.0
(13 rows)

SELECT * FROM versions WHERE version <  '1.5.0';
 version 
---------
 1.0.0
 0.0.0
 1.1.0
 1.2.0
(4 rows)

SELECT * FROM versions WHERE version <= '1.5.0';
 version 
---------
 1.0.0
 0.0.0
 1.1.0
 1.2.0
 1.5.0
 1.05.0
 1.005.0
(7 rows)

SELECT * FROM versions WHERE version =  '2.0.0-alpha';
   version   
-------------
 2.0.0-alpha
(1 row)

SELECT * FROM versions WHERE version != '2.0.0-alpha';
   version   
-------------
 1.0.0
 0.0.0
 1.1.0
 1.2.0
 1.23.0
 1.23.9
 2.9.0
 2.0.12
 3.0.1
 3.0.2
 3.0.4-beta2
 3.0.4-alpha
 3.0.4-rc1
 1.5.0
 1.05.0
 1.005.0
(16 rows)

SELECT * FROM versions WHERE version <> '2.0.0-alpha';
   version   
-------------
 1.0.0
 0.0.0
 1.1.0
 1.2.0
 1.23.0
 1.23.9
 2.9.0
 2.0.12
 3.0.1
 3.0.2
 3.0.4-beta2
 3.0.4-alpha
 3.0.4-rc1
 1.5.0
 1.05.0
 1.005.0
(16 rows)

SELECT * FROM versions WHERE version >  '2.0.0-alpha';
   version   
-------------
 2.9.0
 2.0.12
 3.0.1
 3.0.2
 3.0.4-beta2
 3.0.4-alpha
 3.0.4-rc1
(7 rows)

SELECT * FROM versions WHERE version >= '2.0.0-alpha';
   version   
-------------
 2.9.0
 2.0.12
 2.0.0-alpha
 3.0.1
 3.0.2
 3.0.4-beta2
 3.0.4-alpha
 3.0.4-rc1
(8 rows)

SELECT * FROM versions WHERE version <  '2.0.0-alpha';
 version 
---------
 1.0.0
 0.0.0
 1.1.0
 1.2.0
 1.23.0
 1.23.9
 1.5.0
 1.05.0
 1.005.0
(9 rows)

SELECT * FROM versions WHERE version <= '2.0.0-alpha';
   version   
-------------
 1.0.0
 0.0.0
 1.1.0
 1.2.0
 1.23.0
 1.23.9
 2.0.0-alpha
 1.5.0
 1.05.0
 1.005.0
(10 rows)

SELECT * FROM versions WHERE version > '2.9-beta1';
   version   
-------------
 2.9.0
 3.0.1
 3.0.2
 3.0.4-beta2
 3.0.4-alpha
 3.0.4-rc1
(6 rows)

DROP TABLE IF EXISTS versions;
